kv/tikv/storage/exec/scheduler.go

```diff
@@ -12,54 +12,76 @@ type Sequential struct {
 	queue       chan task
 }
 
+// task is a task to be run by a scheduler.
 type task struct {
 	cmd           tikv.Command
 	resultChannel chan<- tikv.RespResult
 }
 
+// NewSeqScheduler creates a new sequential scheduler.
 func NewSeqScheduler(innerServer tikv.InnerServer) *Sequential {
 	sched := &Sequential{innerServer, make(chan task)}
 	go sched.handleTask()
 	return sched
 }
 
+// handleTask takes a task from Sequential's queue and executes it.
 func (seq *Sequential) handleTask() {
 	for {
 		task := <-seq.queue
 
+		// A nil task is a command to stop and close down the scheduler.
 		if task.cmd == nil && task.resultChannel == nil {
 			close(seq.queue)
 			return
 		}
 
-		reader, err := seq.innerServer.Reader(task.cmd.Context())
-		if err != nil {
-			if regResp := task.cmd.RegionError(tikv.ExtractRegionError(err)); regResp != nil {
-				task.resultChannel <- tikv.RespOk(regResp)
-			} else {
-				task.resultChannel <- tikv.RespErr(err)
-			}
+		// Get the data we need for execution.
+		ctxt := task.cmd.Context()
+		// TODO reader should be a snapshot of the underlying data at a given timestamp, not a generic reader.
+		reader, err := seq.innerServer.Reader(ctxt)
```

Comment:
`Reader` should be closed after used. For example in standalone mode, the `Reader` wraps a `badger.Txn` inside. After a transaction, we should close it by calling `Commit` or `Discard`. Maybe we could add a method named: `Close` into `DBReader` interface to implement this.